### PR TITLE
cosmos in pr preview

### PR DESCRIPTION
**Pull request by vkozio**
## Summary by CodeRabbit

- **New Features**
	- Enhanced pull request preview deployment with improved responsiveness to events.
	- Added a step to upload build artifacts for better organization during deployment.
- **Bug Fixes**
	- Conditional checks implemented to prevent execution of steps when a pull request is closed.
- **Chores**
	- Updated various dependencies to ensure compatibility and access to the latest features.
- **Improvements**
	- Streamlined deployment workflow for the Cosmos application, enhancing clarity and efficiency.

### CODE REVIEW

File: .github/workflows/pr-preview.yml [hunk 1]
```diff
@@ -1,71 +1,87 @@
...
-          restore-keys: |
-            ${{ runner.os }}-pnpm-store-
+          version: 9
 
       - name: Install dependencies
         run: pnpm install
 
-      - name: Apply domain changes in config files
+      - name: Apply domain changes
+        if: github.event.action != 'closed'
```
Comment (dqunbp) [inline .github/workflows/pr-preview.yml:25]: Can we use it once at the job level?

Comment (alesiahil) [reply .github/workflows/pr-preview.yml:25]: @vkozio ping!

File: .github/workflows/pr-preview.yml [hunk 2]
```diff
@@ -1,71 +1,87 @@
...
+        with:
+          surge_token: ${{ secrets.SURGE_TOKEN }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          dist: dist
+          build: echo "Skip build"
+          failOnError: 'true'
+          teardown: 'true'
 
-      - name: Enable surge.sh redirects
-        run: cp dist/index.html dist/200.html
```
Comment (dqunbp) [inline .github/workflows/pr-preview.yml:62]: It better to keep this, to allow restarts from non root page routes without loosing the app

Comment (alesiahil) [reply .github/workflows/pr-preview.yml:62]: @vkozio ping!

**Pull request closed without merging**